### ISSUE TITLE
Specifying -std on C Flags breaks linting

### ISSUE THREAD

**Issue by Crocmagnon** [hebaishi/linter-gcc#8]
When I specify the standard I want to lint against in the `Gcc Default C Flags`, the whole linter doesn't lint anything anymore.

I used this syntax : 
```
-Wall -std=c90
```

With only `-Wall`, a missing semi-colon shows up as an error. When adding `-std=c90`, everything is reported as fine.

I don't have any code to test for C++.

**Comment by hebaishi** [comment]
Hello, and sorry for the delay. I've just checked this with a small C source file that has a missing semicolon, and it seems to be working fine. Could you please provide the console output? linter-gcc writes all command-line options to the console, so that should help us work out what the problem is.

**Comment by Crocmagnon** [comment]
Here is a little screenshot I took after saving the file :
![alt][1]

As you can see, it shows no error, despite missing the semi-colon at the end of line 23.

Here is the console output :
```javascript
/usr/bin/g++
["-Wall", "-std=c90", "-fmax-errors=0", "/Users/gaugendre/Projects/c/agp/00-example/src/main.c"]
```

[1]: https://www.dropbox.com/s/phjtdcqvrrikwdj/Capture%20d%27%C3%A9cran%202015-09-28%2015.05.51.png?dl=0&raw=1

**Comment by hebaishi** [comment]
Interesting. Could you run the command in the terminal and report the output?
The full command would be:
```
/usr/bin/g++ -Wall -std=c90 -fmax-errors=0 /Users/gaugendre/Projects/c/agp/00-example/src/main.c
```

**Comment by hebaishi** [comment]
This is what I get when I run it on Linux:
```
cc1plus: warning: command line option ‘-std=c90’ is valid for C/ObjC but not for C++ [enabled by default]
```
I think the problem is that you're using ```g++``` with that parameter. If you use ```g++ -std=c99``` or ```gcc -std=c90```, everything works fine.

**Comment by Crocmagnon** [comment]
You got it !
Indeed, when I set the exec path to `/usr/bin/gcc`, the linting works as expected even when specifying C standards older than C99.

Could it be a way of improvement ?
For example, having a per-project exec path, or auto-detection of which to use based on project sources (if all are `.c`, then use `gcc`, else use `g++`) ?
My preference goes to the second option.

We already have the option of differentiating flags based on language, does it mean that you already detect the language, so it may be easy to implement a second field for `gcc` as default C compiler ?

I think that what first induced me in error is the wording : `Gcc Default C Flags`, `Gcc Error Limit`, ... It made me think it would use GCC, even it was clearly stated that the extension defaults to G++.

Thanks for your help !

**Comment by Crocmagnon** [comment]
Hm. There is another catch : When I set `-std=c99`, it doesn't work.
I have to use `-std=c++99`, even in the `Gcc Default C Flags`.

**Comment by hebaishi** [comment]
Hello. Regarding, the ```-std=c++99```, this is probably speciic to the implementation of gcc/g++ that you are using, because ```-std=c99``` works with gcc/g++ on linux. Also, you can have project-specific settings. Information on how to do this is on the repository page [here](https://github.com/hebaishi/linter-gcc#project-specific-settings). You will also find it on the extension page on atom.io and inside atom itself.

**Comment by Crocmagnon** [comment]
Just curiosity : which Linux distribution are you running ?

You're right :
```console
$ g++ -std=c99 src/main.c
clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated
error: invalid argument '-std=c99' not allowed with 'C++/ObjC++'
```

Thanks for the tip on project-specific settings! I didn't correctly read the manual, my bad.
Thanks again for your help!

**Comment by hebaishi** [comment]
I'm running Ubuntu 14.04. Glad the problem has been resolved. Closing the issue.
